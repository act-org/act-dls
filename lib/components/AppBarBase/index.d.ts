/**
 * @prettier
 */
import * as React from 'react';
import { AppBarProps } from '@material-ui/core';
export declare type Props = AppBarProps;
declare const AppBarBase: React.FC<Props>;
export default AppBarBase;
//# sourceMappingURL=index.d.ts.map