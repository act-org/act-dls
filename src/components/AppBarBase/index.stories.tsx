/**
 * @prettier
 */

import * as React from 'react';
import { select } from '@storybook/addon-knobs';

import AppBarBase from '.';

export const Base = (): React.ReactElement<any> => (
  <AppBarBase
    color={select(
      'Color',
      {
        Default: 'default',
        Inherit: 'inherit',
        Primary: 'primary',
        Secondary: 'secondary',
        Transparent: 'transparent',
      },
      'primary',
    )}
    position={select(
      'Position',
      {
        Absolute: 'absolute',
        Fixed: 'fixed',
        Relative: 'relative',
        Static: 'static',
        Sticky: 'sticky',
      },
      'fixed',
    )}
  >
    App Bar Here
  </AppBarBase>
);

export default {
  component: AppBarBase,
  parameters: {
    info: {
      text: 'This is the base variant of the AppBar component.',
    },
  },
  title: 'Atoms/AppBar',
};
