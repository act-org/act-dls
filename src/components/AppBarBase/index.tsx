/**
 * @prettier
 */

import * as React from 'react';
import { AppBar, AppBarProps } from '@material-ui/core';

export type Props = AppBarProps;

const AppBarBase: React.FC<Props> = (props: Props): React.ReactElement<any> => (
  <AppBar {...props} />
);

export default AppBarBase;
