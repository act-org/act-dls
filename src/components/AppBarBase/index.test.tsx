/**
 * @prettier
 */

import * as React from 'react';

import render from '~/helpers/test/render';

import AppBarBase from '.';

describe('AppBarBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <AppBarBase color="primary" position="fixed">
        AppBar Base
      </AppBarBase>,
    );

    expect(container).toMatchSnapshot();
  });
});
