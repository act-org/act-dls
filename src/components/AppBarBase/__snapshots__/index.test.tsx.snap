// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBarBase matches the snapshot 1`] = `
<div>
  <button
    class="MuiAppBarBase-root MuiAppBar-root MuiAppBar-text"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiAppBar-label"
    >
      AppBar Label
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
